### ISSUE TITLE
Exponent operator should not trigger when operands are different type and `strictBinaryOperands` is enabled

### ISSUE THREAD

**Issue by zerkms** [vimeo/psalm#5250]
```php
$v = 1.5 ** 10;
```

This code triggers

```
ERROR: InvalidOperand - src.php:131:22 - Cannot add ints to floats (see https://psalm.dev/058)
            $sleep = 1.5 ** 10;
```

while I believe it makes very little sense to demand both operands of `**` be of the same type.

**Comment by weirdan** [comment]
Note: requires strict binary operands option to be enabled.

**Comment by theodorejb** [comment]
This is a big issue when attempting to upgrade a project to Psalm 6, since this option is enabled by default.
https://psalm.dev/r/a7932278f5

Of course I could suppress all `InvalidOperand` errors, but I think they would be helpful when the types are something other than float and int.
